components/Nav/SearchWithAlgolia.js

```diff
@@ -0,0 +1,34 @@
+import React, { Component } from 'react'
+import Router from 'next/router'
+
+import Search from './Search'
+
+let docsearch
+if (process.browser) {
+  docsearch = require('docsearch.js')
+}
+
+class SearchWithAlgolia extends Component {
+  isDocs = true
+
+  componentWillMount() {
+    this.isDocs = process.browser && Router.pathname.startsWith('/docs')
+  }
+
+  componentDidMount() {
+    if (process.browser && typeof docsearch !== 'undefined') {
```

Comment:
Let's require `docsearch` down here instead and remove the `process.browser` stuff since `componentDidMount` doesn't run on the server